docs/tutorial/process-model.md

```diff
@@ -0,0 +1,203 @@
+to understand is:
+
+* An HTML file is your entry point for the renderer process.
+* UI styling is added through Cascading Style Sheets (CSS).
+* Executable JavaScript code can be added through `<script>` elements.
+
+Moreover, this also means that the renderer has no direct access to `require`
+or other Node.js APIs. In order to import NPM modules in the renderer, you must use
+the same bundler toolchains (for example, `webpack` or `parcel`) that you use for web
+apps.
+
+> Note: Renderer processes can be spawned with full Node.js environment for ease of
+> development. Historically, this used to be the default, but this feature was disabled
+> for security reasons.
+
+At this point, you might be wondering how your renderer process user interfaces
+can interact with Node.js and Electron's native desktop functionality if these
+features are only accessible from the main process. In fact, there is no direct
+way to even import Electron's content scripts.
+
+## Preload scripts
+
+<!-- Note: This guide doesn't take sandboxing into account, which might fundamentally 
+change the statements here. -->
+Preload scripts contain code that executes in a renderer process before its web content
+begins loading. These scripts runs within the renderer context, but are granted more
+privileges by having access to Node.js APIs.
+
+A preload script can be attached to the main process in the `BrowserWindow` constructor's
+`webPreferences` option.
+
+```js title='main.js'
+const { BrowserWindow } = require('electron')
+//...
+const win = new BrowserWindow({
+  preload: 'path/to/preload.js'
+})
+//...
+```
+
+Because the preload script shares a global [`Window`][window-mdn] interface with the
+renderers and can access Node.js APIs, it serves to enhance your renderer by exposing
+arbitrary APIs in the `window` global that your web contents can then consume.
+
+Although preload scripts share a `window` global with the renderer they're attached to,
+you cannot directly attach any variables from the preload to `window` because of
+the [`contextIsolation`][context-isolation] default.
+
+```js title='preload.js'
+window.myAPI = {
+  desktop: true
+}
+```
+
+```js title='renderer.js'
+console.log(window.myAPI)
+// => undefined
+```
+
+Context Isolation means that preload scripts are isolated from the renderer's main world
+to avoid leaking any privileged APIs into your web content's code.
+
+Instead, use the [`contextBridge`][context-bridge] module to accomplish this
+securely:
+
+```js title='preload.js'
+const { contextBridge } = require('electron')
+
+contextBridge.exposeInMainWorld('myAPI', {
+  desktop: true
+})
+```
+
+```js title='renderer.js'
+console.log(window.myAPI)
+// => { desktop: true }
+```
+
+This feature is incredibly useful for two main purposes:
+
+* By exposing [`ipcRenderer`][ipcRenderer] helpers to the renderer, you can use
+  inter-process communication (IPC) to trigger main process tasks from the
+  renderer (and vice-versa).
```

Comment:
One day we can link to the IPC tutorial here...

Reply:
That's next up on my list!